elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/AsyncProperties.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.spring.config;
+
+import lombok.Data;
+
+/**
+ * Extra properties for setting up async query support.
+ */
+@Data
+public class AsyncProperties extends ControllerProperties {
+
+    /**
+     * Default thread pool size.
+     */
+    private int threadPoolSize = 6;
+
+    /**
+     * Default max query run time.
+     */
+    private int maxRunTime = 60;
+
+    /**
+     * Default number of hosts running Elide.
+     */
+    private int numberOfHosts = 1;
```

Comment:
Do we need this?

Reply:
so we are using this param to randomize the start-up time for cleanup executor. If we have more than 1 hosts running and then if all of them have the cleanup service executing at the same time they could possibly try to update/delete the record at the same time and have lock error etc.  so this is how we are using it here. https://github.com/yahoo/elide/blob/c88d08a94c994f16ee51c6a51c39694411fb8c59/elide-async/src/main/java/com/yahoo/elide/async/service/AsyncExecutorService.java#L59-L63